### PR TITLE
Feature/template improvements

### PR DESCRIPTION
**Pull request by aorfanos**
Allows use of a custom alert message template file. 

Example invocation would be: `-alert-template-file=alertfile.md` .

I have commented out `Test_formatIssueBody` because I'm still working my way to make it work.

### CODE REVIEW

**Comment by Limess** [comment]
We just started looking at using this project, this feature is a must have though - it'd be great to see this merged and released so we don't have to build and maintain a fork.

**Comment by stephen-soltesz** [comment]
Thank you for this change @aorfanos  - I'm taking a look now. I'm not sure how we missed it this long.

**Comment by aorfanos** [comment]
I've forgotten about it as well - better late than never 😄

**Review by stephen-soltesz — COMMENTED** [review]
If you still have a moment, a few small tweaks and complete unit test coverage, and I'd be happy to accept. (You should be able to access reviewable for replies to specific comments). Please let me know when I should take another look.

> Reviewed 3 of 6 files at r1, 1 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/54#-:-NJvb5ARE2yiKDH-lU6b:b-nogheo)* status: 0 of 1 LGTMs obtained

___
*[`alerts/handler.go` line 184 at r2](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/54#-NJvX68C976Ui5Uf6jF2:-NJvX68C976Ui5Uf6jF3:b-aijpgb) ([raw file](https://github.com/m-lab/alertmanager-github-receiver/blob/c1e9de41e92625f2d97e2cbd04a7400163ed35fd/alerts/handler.go#L184)):*
> ```Go
> 			msgBody, err := rh.formatIssueBody(msg)
> 			if err != nil {
> 				return fmt.Errorf("format body for %q: %s", msg.GroupKey, err)
> ```

Is it possible to get unit test coverage for this error condition?
___
*[`alerts/handler_test.go` line 247 at r2](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/54#-NJvYHyAECZBaY24IaBL:-NJvYHyAECZBaY24IaBM:b-nvy4k8) ([raw file](https://github.com/m-lab/alertmanager-github-receiver/blob/c1e9de41e92625f2d97e2cbd04a7400163ed35fd/alerts/handler_test.go#L247)):*
> ```Go
> 				},
> 			},
> 			titleTmpl: 	   `{{ (index .Data.Alerts 1).Status }}`,
> ```

Please run `go fmt` on all files.
___
*[`alerts/template_test.go` line 62 at r2](https://reviewable.io/reviews/m-lab/alertmanager-github-receiver/54#-NJvYmwz7PomuFXOSQ-X:-NJvYmwz7PomuFXOSQ-Y:b-qom6jn) ([raw file](https://github.com/m-lab/alertmanager-github-receiver/blob/c1e9de41e92625f2d97e2cbd04a7400163ed35fd/alerts/template_test.go#L62)):*
> ```Go
> 
> 			body, err := rh.formatIssueBody(&msg)
> 			if tc.expectErrTxt == "" && err != nil {
> ```

Please simplify the conditional logic verifying the test; I see this is like the one below (it's on me for approving originally! I'll fix it after this change) but I find both hard to follow.

I recommend the (slightly modified) pattern auto-generated by VSCode's "Generate Unit Tests" for this function in Go.

Whether there was an error is sufficient; inspecting the error string content is not necessary imo. We know it will be due to the template execution by inspection.

For example:

```
func Test_formatIssueBody(t *testing.T) {
	tests := []struct {
		name     string
		template string
		want     string
		wantErr  bool
	}{
		{
			name:     "success",
			template: "foo",
			want:     "foo",
			wantErr:  false,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := formatIssueBody(tt.args.msg)
			if (err != nil) != tt.wantErr {
				t.Errorf("formatIssueBody() error = %v, wantErr %v", err, tt.wantErr)
				return
			}
			if got != tt.want {
				t.Errorf("formatIssueBody() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

**Comment by stephen-soltesz** [comment]
@aorfanos thank you for this change - the logic here was the basis of https://github.com/m-lab/alertmanager-github-receiver/pull/59 now merged (your contribution is acknowledged in the commit). We'll prepare a new release this week. I'll close this PR since the functionality is now merged to the main branch.
